File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hash = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int cur = nums[i];
            if (hash.containsKey(target - cur)){
                return new int[]{hash.get(target - cur), i};
            } else {
                hash.put(cur, i);
            }
        }
        return null;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        def go(cur: int, ind: int, out: List[int]):
            for i in range(ind, len(nums)):
                n = nums[i]
                if n + cur == target:
                    out.append(n)
                    final.append(out[:])
                    out.pop()
                elif n + cur < target:
                    out.append(n)
                    go(n + cur, i, out[:])
                    out.pop()
        final = []
        go(0, 0, [])
        return final

